### DISCUSSION TITLE
How to use prettier with editorconfig in project.yml task ?

### DISCUSSION THREAD

**Discussion by tomdavidson** [moonrepo/moon#358] [Q&A] [answered]
`prettier` by default recognizes .editorconfig` to help keep the two from conflicting.  When I run prettier as a moon task the `.editorconfig` is not used in the workspace. 

As a work around, I can repeat all the relevant `.editorconfig` parameters in my prettier configuration, but this seems pretty unfortunate. Perhaps I can run prettier with a root level context with a per workspace path?  I think I could also run prettier as a root level task but filter on touched files in ci to fast ?

Recommendations to make it jive with moon better?

**Comment by milesj** [comment]
@tomdavidson Looks like prettier doesn't support a CLI option to the .editorconfig file and probably assumes it exists in the current working directory. https://prettier.io/docs/en/cli.html

I think your best option here is to always run prettier from the workspace root, instead of in each project.  https://moonrepo.dev/docs/config/project#runfromworkspaceroot

**Comment by tomdavidson** [comment, marked as answer]
Answer: Do not use the `--config` arg. Let prettier find its own config in project root and then it will also find the .editorconfig file.

Thanks @milesj   I just learned that prettier will find the project root in search of the config file and .editorconfig  - [resolve-config-editorconfig.js](https://github.com/prettier/prettier/blob/98f560839cc34b60cb0db192603c7629d0a23b52/src/config/resolve-config-editorconfig.js) - which must be why I haven't ran into this before. 

If I do not provide the argument to the task, i.e.  remove `--config @in(0) ` , prettier still finds the project root, the standard config location, and the .editorconfig. Not sure what it is about how prettier finds the root and the moon task, but this is a pretty great 'workaround' for me.

**Reply by milesj** [reply]
Glad it worked.

This is also good to know. I'll add it to the prettier docs.
